cmd/main.go

```diff
@@ -1,79 +1,201 @@
 // SPDX-License-Identifier: Apache-2.0
 // Copyright (c) 2022-2023 Intel Corporation, or its subsidiaries.
 // Copyright (c) 2022-2023 Dell Inc, or its subsidiaries.
+// Copyright (C) 2023 Nordix Foundation.
 
 // Package main is the main package of the application
 package main
 
 import (
 	"context"
-	"flag"
 	"fmt"
+	"io"
 	"log"
 	"net"
 	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
 	"time"
 
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
 	pc "github.com/opiproject/opi-api/inventory/v1/gen/go"
 	pe "github.com/opiproject/opi-api/network/evpn-gw/v1alpha1/gen/go"
 	"github.com/opiproject/opi-evpn-bridge/pkg/bridge"
+	"github.com/opiproject/opi-evpn-bridge/pkg/config"
+	"github.com/opiproject/opi-evpn-bridge/pkg/infradb"
+	"github.com/opiproject/opi-evpn-bridge/pkg/infradb/taskmanager"
 	"github.com/opiproject/opi-evpn-bridge/pkg/port"
 	"github.com/opiproject/opi-evpn-bridge/pkg/svi"
 	"github.com/opiproject/opi-evpn-bridge/pkg/utils"
 	"github.com/opiproject/opi-evpn-bridge/pkg/vrf"
 	"github.com/opiproject/opi-smbios-bridge/pkg/inventory"
 
-	"github.com/philippgille/gokv"
-	"github.com/philippgille/gokv/redis"
-
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/insecure"
 	"google.golang.org/grpc/reflection"
 
 	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/logging"
 	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
-
+	ci_linux "github.com/opiproject/opi-evpn-bridge/pkg/LinuxCIModule"
+	gen_linux "github.com/opiproject/opi-evpn-bridge/pkg/LinuxGeneralModule"
+	ipu_linux "github.com/opiproject/opi-evpn-bridge/pkg/LinuxVendorModule/ipu"
+	frr "github.com/opiproject/opi-evpn-bridge/pkg/frr"
 	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
 )
 
-func main() {
-	var grpcPort int
-	flag.IntVar(&grpcPort, "grpc_port", 50151, "The gRPC server port")
+const (
+	configFilePath = "./"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "opi-evpn-bridge",
+	Short: "evpn bridge",
+	Long:  "evpn bridge application",
+	PreRunE: func(_ *cobra.Command, _ []string) error {
+		return validateConfigs()
+	},
+	Run: func(_ *cobra.Command, _ []string) {
+
+		taskmanager.TaskMan.StartTaskManager()
 
-	var httpPort int
-	flag.IntVar(&httpPort, "http_port", 8082, "The HTTP server port")
+		err := infradb.NewInfraDB(config.GlobalConfig.DBAddress, config.GlobalConfig.Database)
+		if err != nil {
+			log.Println("error in creating db", err)
+		}
+		go runGatewayServer(config.GlobalConfig.GRPCPort, config.GlobalConfig.HTTPPort)
 
-	var tlsFiles string
-	flag.StringVar(&tlsFiles, "tls", "", "TLS files in server_cert:server_key:ca_cert format.")
+		defer func() {
+			if err := infradb.Close(); err != nil {
+				log.Fatal(err)
+			}
+		}()
 
-	var redisAddress string
-	flag.StringVar(&redisAddress, "redis_addr", "127.0.0.1:6379", "Redis address in ip_address:port format")
+		switch config.GlobalConfig.Buildenv {
+		case "ipu":
```

Comment:
This code is generic enough and can be used for different IPU/dpus that you have underneath and as well for the CI env where you have no IPU/DPU. Here the case : ipu is when you have an IPU smartnic underneath. Even though that the code is generic some modules need to be provided by the vendor to address the specific smartnic that you have underneath so with this case: ipu we capture the case and we load some of the specific modules that are realted to the IPU card. If for instance you had a dpu bluefield as a card the vendor will need to provide some specific modules for that card and then we will create a case dpu bluefield for that specific card where the ppropriate modules will be loaded

Reply:
I understand. I mean, let's change the name and use more widespread terminology, like `"dpu"`

Reply:
I do not think this is clear. According to the card that you have underneath you need to load certain modules. If you have an intel ipu for example you need to load the ipu_linux module. If you have a Marvell card you need to load the Marvel_linux. If you have a Bluefield you need to load a bluefield_linux module. So this is what this case does it loads the correct module related to the card that you have. So for intel IPU we have an ipu_linux module that we need to load. We cannot just change that to dpu because dpu can be Marvell or Bluefield. BTW by ipu here we are talking only for an Intel IPU. So the IPU case is pretty unique but the dpu case it isn't because you can have a Marvell dpu or a Bluefield dpu.

Is that more clear now ?

Reply:
I've heard that the term ipu is also used to other dpus/xpus from different vendors within Intel.
Then let's be more specific and name it `ipu_mev`/`mev` since the description says Mount Evans or `ipu_intel`?

Reply:
yes ok let me think about this a bit

Reply:
Hello @artek-koltun what do you think about intel_e2000?  E2000 is the part name for a mount evans. So this means that all the vendor related files in the code will have the same name.

If you agree with this change we can start do the changes on the naming.

Reply:
`intel_e2000` sounds good

Reply:
Hello @atulpatel261194 Can you please update where is needed the name ipu with the name intel_e2000 ? Not only updating the config but also the golang filenames.

Reply:
Done